packages/e2e-test-utils/src/inserter.js

```diff
@@ -103,7 +103,13 @@ async function waitForInserterCloseAndContentFocus() {
  * Wait for the inserter search to yield results because that input is debounced.
  */
 async function waitForInserterSearch() {
-	await page.waitForSelector( '.block-editor-inserter__no-tab-container' );
+	try {
```

Comment:
Potentially we could use `Promise.any` and only wait like 2 seconds or so

Reply:
True, we could. We could also reduce the timeout of `waitForSelector()` to like 2 seconds, which should be more than enough. No need to wait for the full 30 seconds. Done in https://github.com/WordPress/gutenberg/pull/46268/commits/fc8f801e172ac1511319dc96d543f6e9b93d8e34

Reply:
I think a universal solution to this could be using the [`Page.waitForResponse`](https://pptr.dev/api/puppeteer.page.waitforresponse/) method. AFAIR, both previous and current Inserter versions use the `/search` endpoint, so something like this should be more stable and require less maintenance at the same time:

```js
await page.waitForResponse( '/search' );
```

Reply:
To be extra sure the response is not returned before the `waitForInserterSearch` is called, I'd couple it with the input so that we start waiting before the search request is triggered, for example:

```js
async function inserterSearch( searchTerm ) {
	return await Promise.all( [
		page.waitForResponse( '/search' ),
		page.keyboard.type( searchTerm )
	] );
}
```

Puppeteer doesn't seem to be recommending this convention, but [Playwright does](https://playwright.dev/docs/api/class-page#page-wait-for-response). I think both implementations work the same way, so my recommendation would be going with the Playwright convention, also because we're currently migrating to it :)

Reply:
Makes a lot of sense! Thanks for offering a better alternative @WunderBart! Would you like to follow up with that, or should I give it a go?

Reply:
I'll follow up - should have something today!

Reply:
Thanks a bunch, @WunderBart 🙌

Reply:
Sharing an early WiP https://github.com/WordPress/gutenberg/pull/46366 - did a bit more than planned, but it's mostly removing unnecessary waits and code duplication. Waiting for the test results now! 🤞

Reply:
Thanks @WunderBart! Let me know when you feel it's ready for a look.

Reply:
It's basically ready, @tyxla. I'm still looking at the few tests that keep failing, but they don't seem to be related to the change, and they do pass for me locally. 🤔 I've just updated the branch with the current trunk, so hopefully it helps. 🤞

Reply:
Will keep an eye on it 👍